### PR TITLE
Support creating data disk with instance

### PR DESCRIPTION
**Pull request by nightmeng**
TF_DEBUG=debug TF_ACC=1 go  test -v -run=TestAccAlicloudInstanceSecurityEnhancementStrategy_update
=== RUN   TestAccAlicloudInstanceSecurityEnhancementStrategy_update
--- PASS: TestAccAlicloudInstanceSecurityEnhancementStrategy_update (272.63s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     272.678s

TF_DEBUG=debug TF_ACC=1 go  test -v -run=TestAccAlicloudInstance_dataDisk
=== RUN   TestAccAlicloudInstance_dataDisk
--- PASS: TestAccAlicloudInstance_dataDisk (107.06s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     107.117s

### CODE REVIEW

File: alicloud/resource_alicloud_instance.go [hunk 1]
```diff
@@ -121,6 +121,59 @@ func resourceAliyunInstance() *schema.Resource {
...
+						},
+						"category": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validateAllowedStringValue([]string{
+								string(DiskCloud),
+								string(DiskCloudEfficiency),
+								string(DiskCloudSSD),
+								string(DiskEphemeralSSD)}),
+							Default:  "cloud",
```
[the remark this answers is not in the document]
Comment (nightmeng) [reply alicloud/resource_alicloud_instance.go]: Use DiskCloud instead of DiskCloudEfficiency.

File: alicloud/resource_alicloud_instance.go [hunk 2]
```diff
@@ -121,6 +121,55 @@ func resourceAliyunInstance() *schema.Resource {
...
+				Type:     schema.TypeList,
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 15,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_instance.go]: Please add validfunc for `name` and `description` based on the existing resource `alicloud_disk`

File: alicloud/resource_alicloud_instance.go [hunk 3]
```diff
@@ -803,6 +861,38 @@ func buildAliyunInstanceArgs(d *schema.ResourceData, meta interface{}) (*ecs.Cre
...
+	if v, ok := d.GetOk("data_disks"); ok {
+		disks := v.([]interface{})
+		var dataDiskRequests []ecs.CreateInstanceDataDisk
+		for i := range disks {
+			disk := disks[i].(map[string]interface{})
+
+			req := ecs.CreateInstanceDataDisk{
+				Category:           disk["category"].(string),
+				DeleteWithInstance: fmt.Sprintf("%v", disk["delete_with_instance"].(bool)),
+				Encrypted:          fmt.Sprintf("%v", disk["encrypted"].(bool)),
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_instance.go]: Please use `requests.NewBoolean()` instead `fmt.Sprintf("%v", ...)`

File: alicloud/resource_alicloud_instance.go [hunk 4]
```diff
@@ -803,6 +861,38 @@ func buildAliyunInstanceArgs(d *schema.ResourceData, meta interface{}) (*ecs.Cre
...
+			if name, ok := disk["name"]; ok {
+				req.DiskName = name.(string)
+			}
+			if snapshotId, ok := disk["snapshot_id"]; ok {
+				req.SnapshotId = snapshotId.(string)
+			}
+			if description, ok := disk["description"]; ok {
+				req.Description = description.(string)
+			}
+			req.Size = fmt.Sprintf("%d", disk["size"].(int))
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_instance.go]: Please use requests.NewInteger.

File: website/docs/r/instance.html.markdown [hunk 5]
```diff
@@ -119,6 +119,28 @@ However, because of changing instance charge type has CPU core count quota limit
 * `security_enhancement_strategy` - (Optional, Force New) The security enhancement strategy.
     - Active: Enable security enhancement strategy, it only works on system images.
     - Deactive: Disable security enhancement strategy, it works on all images.
+* `data_disks` - (Optional, List, Force New) The list of data disks created with instance.
```
Comment (xiaozhu36) [inline website/docs/r/instance.html.markdown]: Pelase add a tag for the `data_disks`: (Optional, ForceNew, Available 1.24.0+)

File: website/docs/r/instance.html.markdown [hunk 6]
```diff
@@ -119,6 +119,28 @@ However, because of changing instance charge type has CPU core count quota limit
...
+        - cloud_efficiency：[20, 32768]
+        - cloud_ssd：[20, 32768]
+        - ephemeral_ssd：[5, 800]
+    * `category` - (Optional, Force New) The category of the disk:
+        - `cloud`: The general cloud disk.
+        - `cloud_efficiency`: The efficiency cloud disk.
+        - `cloud_ssd`: The SSD cloud disk.
+        - `ephemeral_ssd`: The local SSD disk.
+
+        Default to cloud.
```
Comment (xiaozhu36) [inline website/docs/r/instance.html.markdown]: Please match it with schema

File: alicloud/resource_alicloud_instance.go (hunk 4 above)
Comment (nightmeng) [reply alicloud/resource_alicloud_instance.go]: API required

**Pull request merged**